src/plugins/web.js

```diff
@@ -84,70 +85,16 @@ export default {
             return
           }
 
-          // download without url signing enabled
-          this.addActionToProgress(file)
-          return this.$_downloadFileAsBlob(url, headers, file)
-        },
-        $_downloadFileAsBlob(url, headers, file) {
-          const request = new XMLHttpRequest()
-          request.open('GET', url)
-          request.responseType = 'blob'
-          Object.keys(headers).forEach(p => request.setRequestHeader(p, headers[p]))
-
-          request.addEventListener('readystatechange', () => {
-            if (request.readyState === 2 && request.status !== 200) {
-              this.showMessage({
-                title: this.$gettext('Download failed'),
-                desc: request.statusText,
-                status: 'danger',
-                autoClose: {
-                  enabled: true
-                }
-              })
-
-              // Remove item from progress queue in case the request failed
-              this.removeActionFromProgress(file)
-            } else if (request.readyState === 4 && request.status === 200) {
-              // Download has finished
-              if (window.navigator && window.navigator.msSaveOrOpenBlob) {
-                // for IE
-                window.navigator.msSaveOrOpenBlob(request.response, file.name)
-              } else {
-                // for Non-IE (chrome, firefox etc.)
-                const objectUrl = window.URL.createObjectURL(request.response)
-
-                const anchor = document.createElement('a')
-                anchor.href = objectUrl
-                anchor.download = file.name
-                anchor.style.display = 'none'
-                document.body.appendChild(anchor)
-                anchor.click()
-                document.body.removeChild(anchor)
-                window.URL.revokeObjectURL(objectUrl)
-
-                // TODO: Bring back when progress bar for download is figured out
-                // Items with small size can be fetched too fast for progress listener
-                // This ensures that they'll be removed from progress queue
-                // this.updateFileProgress({
-                //   fileName: file.name,
-                //   progress: 100
-                // })
-              }
-
-              this.removeActionFromProgress(file)
-            }
-          })
-
-          // TODO: Bring back when progress bar for download is figured out
-          // request.addEventListener('progress', e => {
-          //   const progress = (e.loaded / e.total) * 100
-          //   this.updateFileProgress({
-          //     fileName: file.name,
-          //     progress
-          //   })
-          // })
-
-          request.send()
+          const a = document.createElement('a')
```

Comment:
I would like to get rid of this code duplication with the `a` element.

Reply:
Can be moved out into its own method.

Reply:
Should I tackle this here in this PR?

Reply:
Yes please :-)